tests/integration/models/device.spec.ts

```diff
@@ -3985,7 +3952,88 @@ describe('Device Model', function () {
 									'2.29.2+rev1.prod',
 								),
 							).to.not.throw();
-						}));
+						});
+					});
+
+					it('should throw when updating to a pre-release version with an older server', () => {
+						[
+							['balenaOS 2.29.2-1704382618288+rev1', 'prod'],
+							['balenaOS 2.29.2+rev1', 'prod'],
+						].forEach(function ([osVersion, osVariant]) {
+							expect(() =>
+								_checkOsUpdateTarget(
+									{
+										uuid,
+										is_of__device_type: [{ slug: 'raspberrypi3' }],
+										is_online: true,
+										os_version: osVersion,
+										os_variant: osVariant,
+									},
+									'2.28.0-1704382553234+rev1.prod',
+								),
+							).to.throw('OS downgrades are not allowed');
+						});
+					});
+
+					it('should not throw when updating to a pre-release version with a newer base server', () => {
+						[['balenaOS 2.28.0+rev1', 'prod']].forEach(function ([
+							osVersion,
+							osVariant,
+						]) {
+							expect(() =>
+								_checkOsUpdateTarget(
+									{
+										uuid,
+										is_of__device_type: [{ slug: 'raspberrypi3' }],
+										is_online: true,
+										os_version: osVersion,
+										os_variant: osVariant,
+									},
+									'2.29.2-1704382618288+rev1.prod',
```

Comment:
Can we please have comments on why these specific OS versions were chosen? Same on line 3999

Reply:
B/c that's the only one atm on prod that is draft :)

Reply:
Ends up the versions in this file are random, so I added a comment about that right at the start of the `describe()`.